src/pages/Card/_components/SidebarFilter.tsx

```diff
@@ -16,25 +21,61 @@ function SidebarFilter() {
     CHECK_BOX_DATA.map((category) => Array(category.checkboxes.length).fill(false)),
   );
 
-  // 전체보기 체크박스 상태 변경 함수
+  // "전체보기" 체크박스 변경
   const handleCheckChange = () => {
-    setIsAllChecked((prev) => !prev);
+    const newAllChecked = !isAllChecked;
+    setIsAllChecked(newAllChecked);
+    if (newAllChecked) {
+      // "전체보기"가 체크되면 모든 체크박스를 해제하고, 전체카드를 가져옴
+      setCheckboxStates(
+        CHECK_BOX_DATA.map((category) => Array(category.checkboxes.length).fill(false)),
+      );
+      onAllCheck();
+    } else {
+      onAllCheck(); // 전체보기 해제 시에도 전체카드를 다시 가져옴
+    }
   };
 
-  // 드롭다운 카테고리 활성화 상태 변경 함수
+  // 카테고리 클릭
   const handleCategoryClick = (index: number) => {
     setActiveCategories((prev) => prev.map((isActive, i) => (i === index ? !isActive : isActive)));
   };
 
-  // 체크박스 상태 변경 함수
+  // 개별 체크박스 변경
   const handleCheckboxChange = (categoryIndex: number, checkboxIndex: number) => {
-    setCheckboxStates((prevStates) =>
-      prevStates.map((category, catIdx) =>
-        catIdx === categoryIndex
-          ? category.map((checked, chkIdx) => (chkIdx === checkboxIndex ? !checked : checked))
-          : category,
-      ),
-    );
+    setIsAllChecked(false);
+    setCheckboxStates((prevStates) => {
+      const updatedStates = prevStates.map(
+        (category, catIdx) =>
+          catIdx === categoryIndex
+            ? category.map((checked, chkIdx) => (chkIdx === checkboxIndex ? !checked : checked))
+            : Array(category.length).fill(false), // 다른 카테고리 체크박스 해제
+      );
+
+      const selectedTags: string[] = [];
+      let selectedCategory = '';
+
+      // 선택된 태그들을 수집
+      updatedStates.forEach((category, catIdx) => {
+        const categorySelectedTags: string[] = [];
+        category.forEach((checked, chkIdx) => {
+          if (checked) {
+            categorySelectedTags.push(CHECK_BOX_DATA[catIdx].checkboxes[chkIdx].title);
+          }
+        });
+        if (categorySelectedTags.length > 0) {
+          selectedCategory = CHECK_BOX_DATA[catIdx].categoryName;
+          selectedTags.push(...categorySelectedTags);
+        }
+      });
```

Comment:
이쪽에서도 불변성을 지키면서 배열을 업데이트하는건 어떠신가요~~? 불변성을 지키면서 코드를 만들면 나중에 우리도 모르게 발생할 오류를 방지할 수 있어요!

```suggestion
 let selectedTags: string[] = [];
      let selectedCategory = '';

      // 선택된 태그들을 수집
      updatedStates.forEach((category, catIdx) => {
        let categorySelectedTags: string[] = [];
        category.forEach((checked, chkIdx) => {
          if (checked) {
            categorySelectedTags = [...categorySelectedTags, CHECK_BOX_DATA[catIdx].checkboxes[chkIdx].title]
          }
        });
        if (categorySelectedTags.length > 0) {
          selectedCategory = CHECK_BOX_DATA[catIdx].categoryName;
          selectedTags = [...selectedTags, ...categorySelectedTags];
        }
      });
```

Reply:
불변성을 유지는 생각 못했었는데, 대원님 말대로 불변성 지키며 코드짜면 유지 보수성과 성능 측면에서 더 많은 이점을 줄 거 같네요..!! 감사해요!!